apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -565,30 +596,70 @@ def parse(element_html, data):
         element, "grading-method", GRADING_METHOD_DEFAULT
     )
     correct_answers = data["correct_answers"][answer_name]
+    blocks = data["params"][answer_name]
 
     if grading_mode == "ranking":
         for answer in student_answer:
-            search = next(
+            matching_block = next(
                 (
                     item
                     for item in correct_answers
                     if item["inner_html"] == answer["inner_html"]
                 ),
                 None,
             )
-            answer["ranking"] = search["ranking"] if search is not None else None
-            answer["tag"] = search["tag"] if search is not None else None
+            answer["ranking"] = (
```

Comment:
Instead of using inline assignment for these two keys, just wrap this block with a `if matching_block is None` check. You can even combine it with the one below, I think.

Reply:
I would also need an else block to assign none, I think it would just get more complicated that way.